src/main/java/org/openstreetmap/atlas/checks/utility/UniqueCheckFlagContainer.java

```diff
@@ -1,109 +1,90 @@
 import org.openstreetmap.atlas.checks.flag.CheckFlag;
 
 /**
- * A container that will deduplicate check flags based on source and unique IDs
+ * A container used to deduplicate check flags based on checkName and unique IDs
  *
  * @author jklamer
  * @author bbreithaupt
  */
 public class UniqueCheckFlagContainer implements Serializable
 {
 
-    private final ConcurrentHashMap<String, ConcurrentHashMap<Set<String>, CheckFlag>> uniqueFlags;
+    private String checkName;
+    private Set<String> uniqueIdentifiers;
+    private CheckFlag checkFlag;
 
     /**
-     * Combines to containers. This deduplicates {@link CheckFlag}s by overwiting ones with matching
-     * sources and IDs.
-     *
-     * @param container1
-     *            {@link UniqueCheckFlagContainer}
-     * @param container2
-     *            {@link UniqueCheckFlagContainer}
-     * @return merged {@link UniqueCheckFlagContainer}
+     * @param checkFlagEvent
+     *            {@link CheckFlagEvent}
      */
-    public static UniqueCheckFlagContainer combine(final UniqueCheckFlagContainer container1,
-            final UniqueCheckFlagContainer container2)
+    public UniqueCheckFlagContainer(final CheckFlagEvent checkFlagEvent)
     {
-        container2.uniqueFlags.entrySet()
-                .forEach(entry -> container1.addAll(entry.getKey(), entry.getValue().values()));
-        return container1;
+        this(checkFlagEvent.getCheckName(), checkFlagEvent.getCheckFlag().getUniqueIdentifiers(),
+                checkFlagEvent.getCheckFlag().makeComplete());
     }
 
-    public UniqueCheckFlagContainer()
+    /**
+     * @param checkName
+     *            {@link String} check name
+     * @param uniqueIdentifiers
+     *            {@link Set} of {@link String}s from {@link CheckFlag#getUniqueIdentifiers()}
+     * @param checkFlag
+     *            {@link CheckFlag}
+     */
+    public UniqueCheckFlagContainer(final String checkName, final Set<String> uniqueIdentifiers,
+            final CheckFlag checkFlag)
     {
-        this.uniqueFlags = new ConcurrentHashMap<>();
+        this.checkName = checkName;
+        this.uniqueIdentifiers = uniqueIdentifiers;
+        this.checkFlag = checkFlag.makeComplete();
     }
 
-    @SuppressWarnings("s1144")
-    // Ignore unused constructor warning, this is used for deserialization
-    private UniqueCheckFlagContainer(
-            final ConcurrentHashMap<String, ConcurrentHashMap<Set<String>, CheckFlag>> flags)
+    @Override
+    public boolean equals(final Object other)
     {
-        this.uniqueFlags = flags;
+        if (this == other)
+        {
+            return true;
+        }
+        if (other == null || getClass() != other.getClass())
+        {
+            return false;
+        }
+        final UniqueCheckFlagContainer that = (UniqueCheckFlagContainer) other;
+        return Objects.equals(this.checkName, that.checkName)
+                && Objects.equals(this.uniqueIdentifiers, that.uniqueIdentifiers);
```

Comment:
I think this is a good way to do it, but I have run into odd cases where checkflags do not have any atlas objects associated with them. It would be the check's fault, however, something to note if we run into it again and are wondering where all the flags went